src/main/java/core/basesyntax/service/impl/FruitTransaction.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.exceptions.IncorrectOperationException;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.strategy.OperationStrategy;
+
+public class FruitTransaction {
+    private OperationType operationType;
+    private final Fruit fruit;
+    private final OperationStrategy operationStrategy;
```

Comment:
let's remove `operationStrategy` field and call its method for each transaction in another place
let this class have only state